gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java

```diff
@@ -1544,8 +1545,15 @@ public void firstSpawn() {
 
         // Now its time for the join event since the player is fully loaded
         PlayerJoinEvent event = this.getConnection().getNetworkManager().getServer().getPluginManager().callEvent( new PlayerJoinEvent( this ) );
+        event.setJoinMessage( ChatColor.YELLOW + this.getDisplayName() + ChatColor.YELLOW + " joined the game." );
         if ( event.isCancelled() ) {
             this.connection.disconnect( event.getKickReason() );
+        } else {
```

Comment:
Add if join mesage from event is neither null nor empty, otherwise don't do anything

Reply:
Did you completely miss the next line?